monix-reactive/shared/src/main/scala/monix/reactive/internal/builders/Zip2Observable.scala

```diff
@@ -88,8 +88,9 @@ private[reactive] final class Zip2Observable[A1, A2, +R](obsA1: Observable[A1],
           }
       }
 
-      continueP.completeWith(lastAck)
+      val oldP = continueP
       continueP = Promise[Ack]()
+      oldP.completeWith(lastAck)
```

Comment:
This one is a bit puzzling... The method is called from "synchronized" block but I noticed that `onNext` was allowing the next element in between completing `lastAck` and replacing `continueP` with a new promise